sdk/containerregistry/azure-containerregistry/azure/containerregistry/_exchange_client.py

```diff
@@ -47,29 +48,38 @@ def __init__(self, endpoint, credential, **kwargs):
         if not endpoint.startswith("https://") and not endpoint.startswith("http://"):
             endpoint = "https://" + endpoint
         self._endpoint = endpoint
-        self._credential_scopes = "https://management.core.windows.net/.default"
+        self.credential_scope = "https://management.core.windows.net/.default"
         self._client = ContainerRegistry(
             credential=credential,
             url=endpoint,
             sdk_moniker=USER_AGENT,
             authentication_policy=ExchangeClientAuthenticationPolicy(),
-            credential_scopes=kwargs.pop("credential_scopes", self._credential_scopes),
+            credential_scopes=kwargs.pop("credential_scopes", self.credential_scope),
             **kwargs
         )
         self._credential = credential
+        self._refresh_token = None
+        self._last_refresh_time = 0
 
     def get_acr_access_token(self, challenge, **kwargs):
         # type: (str, Dict[str, Any]) -> str
         parsed_challenge = _parse_challenge(challenge)
-        refresh_token = self.exchange_aad_token_for_refresh_token(service=parsed_challenge["service"])
+        refresh_token = self.get_refresh_token(parsed_challenge["service"], **kwargs)
         return self.exchange_refresh_token_for_access_token(
             refresh_token, service=parsed_challenge["service"], scope=parsed_challenge["scope"], **kwargs
         )
 
+    def get_refresh_token(self, service, **kwargs):
+        # type: (str, **Any) -> str
+        if not self._refresh_token or time.time() - self._last_refresh_time > 300:
```

Comment:
I think it's fair but want to confirm: we are assuming that `service` would be the same for a container registry client thus caching is fine?

Reply:
@jeremymeng yes that is the assumption